sdks/java/io/elasticsearch/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOIT.java

```diff
@@ -0,0 +1,153 @@
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.PCollection;
+import org.elasticsearch.client.transport.TransportClient;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A test of {@link ElasticsearchIO} on an independent
+ * Elasticsearch instance.
+ *
+ * <p>This test requires a running instance of Elasticsearch, and the test dataset must exist in the
+ * database.
+ *
+ * <p>You can run this test by doing the following from the beam parent module directory:
+ *
+ * <pre>
+ *  mvn -e -Pio-it verify -pl sdks/java/io/elasticsearch -DintegrationTestPipelineOptions='[
+ *  "--elasticsearchServer=1.2.3.4",
+ *  "--elasticsearchHttpPort=9200",
+ *  "--elasticsearchTcpPort=9300" ]'
+ * </pre>
+ *
+ */
+public class ElasticsearchIOIT {
+  private static final int AVERAGE_DOC_SIZE = 25;
+  private static final Logger LOGGER = LoggerFactory.getLogger(ElasticsearchIOIT.class);
+  private static TransportClient client;
+  private static ElasticsearchTestOptions options;
+  private static ElasticsearchIO.ConnectionConfiguration readConnectionConfiguration;
+  @Rule public TestPipeline pipeline = TestPipeline.create();
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    PipelineOptionsFactory.register(ElasticsearchTestOptions.class);
+    options = TestPipeline.testingPipelineOptions().as(ElasticsearchTestOptions.class);
+    client = ElasticsearchTestDataSet.getClient(options);
+    readConnectionConfiguration =
+        ElasticsearchTestDataSet.getConnectionConfiguration(
+            options, ElasticsearchTestDataSet.ReadOrWrite.READ);
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    ElasticsearchTestDataSet.deleteIndex(client, ElasticsearchTestDataSet.ReadOrWrite.WRITE);
+    client.close();
+  }
+
+  @Test
+  public void testSplitsVolume() throws Exception {
+    ElasticsearchIO.Read read =
+        ElasticsearchIO.read().withConnectionConfiguration(readConnectionConfiguration);
+    ElasticsearchIO.BoundedElasticsearchSource initialSource =
+        new ElasticsearchIO.BoundedElasticsearchSource(read, null);
+    long desiredBundleSizeBytes = 0;
+    List<? extends BoundedSource<String>> splits =
+        initialSource.splitIntoBundles(desiredBundleSizeBytes, options);
+    SourceTestUtils.assertSourcesEqualReferenceSource(initialSource, splits, options);
+    long expectedNumSplits = 5;
+    assertEquals(expectedNumSplits, splits.size());
+    int nonEmptySplits = 0;
+    for (BoundedSource<String> subSource : splits) {
+      if (readFromSource(subSource, options).size() > 0) {
+        nonEmptySplits += 1;
+      }
+    }
+    assertEquals(expectedNumSplits, nonEmptySplits);
+  }
+
+  @Test
+  @Category(RunnableOnService.class)
```

Comment:
I tested without @Category(RunnableOnService.class), and it works, so I removed it. Did not know that.Thanks